baseclasses/decorators.py

```diff
@@ -0,0 +1,26 @@
+import functools
+import unittest
+
+
+def require_mpi(func):
+    try:
+        from mpi4py import MPI  # noqa
+
+        # this is a wrapper on func
+        # but we inject the MPI option in the signature
+        def wrapper(*args, **kwargs):
+            func(*args, **kwargs)
+
+        return wrapper
+    except ImportError:
+        msg = "mpi4py is not installed."
+        if not isinstance(func, type):
+            # this wraps obj, which is the actual function
+            @functools.wraps(func)
+            def skip_wrapper(*args, **kwargs):
+                raise unittest.SkipTest(msg)
+
+            obj = skip_wrapper
+        obj.__unittest_skip__ = True
```

Comment:
This signature replaces `unittest.skip` decorator, right? Why do we both raise `unittest.SkipTest` and add the extra attributes to `obj`?

Reply:
This is also an old commit, take a look at the full diff.